libraries/AP_GPS/AP_GPS_UBLOX.cpp

```diff
@@ -979,17 +979,25 @@ AP_GPS_UBLOX::_parse_gps(void)
                         gnssCount++;
                     }
                 }
-
                 for(int i = 0; i < _buffer.gnss.numConfigBlocks; i++) {
-                    // Reserve an equal portion of channels for all enabled systems
+                    // Reserve an equal portion of channels for all enabled systems that supports it
                     if(gps._gnss_mode[state.instance] & (1 << _buffer.gnss.configBlock[i].gnssId)) {
-                        if(GNSS_SBAS !=_buffer.gnss.configBlock[i].gnssId) {
+
+                        if(GNSS_SBAS !=_buffer.gnss.configBlock[i].gnssId && (supports_F9_config() || is_ublox_M9() || GNSS_GALILEO !=_buffer.gnss.configBlock[i].gnssId)) {
```

Comment:
```suggestion
                        if(GNSS_SBAS !=_buffer.gnss.configBlock[i].gnssId && (_hardware_generation > UBLOX_M8 || GNSS_GALILEO !=_buffer.gnss.configBlock[i].gnssId)) {
```

Why not structure it more like this? The problem with the f9 config/m9 config is you are comitting to a course that requires future updates everytime a new module is shipped, whereas if you just check for anything newer/older then the M8 as your baseline point.... I don't think we benefit by proliferating helpers into this.